docs/content/concepts/sui-move-concepts/conventions.mdx

```diff
@@ -0,0 +1,331 @@
+---
+title: Conventions
+description: Recommended Sui Move 2024 best practices
+---
+
+The recommendations below are based on 2024 Move.
+
+## Add section titles
+
+```move
+module conventions::comments {
+  // === Imports ===
+
+  // === Constants ===
+
+  // === Errors ===
+
+  // === Structs ===
+
+  // === Public-View Functions ===
+
+  // === Public-Mutative Functions ===
+
+  // === Public-Friend Functions ===
+
+  // === Admin Functions ===
+
+  // === Private Functions ===
+
+  // === Test Functions ===
+}
+```
+
+## CRUD functions names
+
+- `add`: Adds a value.
+- `new`: Creates an object.
+- `drop`: Drops a struct.
+- `empty`: Creates a struct.
+- `remove`: Removes a value.
+- `exists`: Checks if a key exists.
+- `contains`: Checks if a collection contains a value.
+- `property_name`: Reads a property.
+- `destroy_empty`: Destroys an object or data structure that has values with the **drop** ability.
+- `to_object_name`: Transforms an Object X to Object Y.
+- `from_object_name`: Transforms an Object Y to Object X.
+- `property_name_mut`: Returns a mutable reference.
+- `property_name_immut`: Returns an immutable reference.
+
+## Potato Structs
+
+#### Do not call structs as Potato. It is a pattern recognized by the lack of abilities.
+
+```move
+module conventions::request {
+  // ✅ Right
+  struct Request {}
+
+  // ❌ Wrong
+  struct RequestPotato {}
+}
+```
+
+## Read Functions
+
+#### Be mindful of the dot syntax when naming functions. Avoid using the object name on function names.
+
+```move
+module conventions::lib {
+  struct Profile {
+    age: u64
+  }
+
+  // ✅ Right
+  public fun age(self: &Profile):  u64 {
+    self.age
+  }
+
+  // ❌ Wrong
+  public fun profile_age(self: &Profile): u64 {
+    self.age
+  }
+}
+
+module conventions::lib {
+  use conventions::lib::{Self, Profile};
+
+  public fun get_tokens(profile: &Profile) {
+
+  // ✅ Right
+  let name = profile.age();
+
+  // ❌ Wrong
+  let name2 = profile.profile_age();
+}
+```
+## Empty Function
+
+#### Functions that create data structures must be called empty.
+
+```move
+module conventions::data_structure {
+  struct DataStructure has copy, drop, store {
+    bits: vector<u8>
+  }
+
+  public fun empty(): DataStructure {}
+}
+```
+## New Function
+
+#### Functions that create objects must be called new.
+
+```move
+module conventions::object {
+  struct Object has key, store {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Object {}
```

Comment:
Would be great to make it either compile or mark that there's a missing implementation.

Suggestions:
```
public fun new(ctx: &mut TxContext): Object {  /* ... */ }

public fun new(ctx: &mut TxContext): Object {
  abort ENotImplemented
}
```